portal/models/fhir.py

```diff
@@ -526,11 +567,31 @@ def consolidate_answer_pairs(answers):
 
         return filtered_answers
 
+    def entry_method(row_data, qnr_data):
+        # Todo: replace with EC.PAPER CodeableConcept
+        if 'paper' in (c.get('code') for c in qnr_data['encounter']['type']):
+            return 'enter manually - paper'
+        if row_data.get('subject_id') == row_data.get('author_id'):
+            return 'online'
+        else:
```

Comment:
As the UI is implemented, it wouldn't be possible for a patient to enter their own paper version (subject_id == author_id). It is a little fragile but I hope to lean more on Encounter(.type) to improve it.